control_panel_api/models.py

```diff
@@ -59,6 +59,23 @@ def aws_create_role(self):
     def aws_delete_role(self):
         services.delete_role(self.aws_role_name)
 
+    @property
+    def _repo_name(self):
+        '''
+        Returns the repo name
+
+        The name is the part after the last slash in the URL, without
+        the '.git' (if present).
+
+        Examples:
+
+        "https://github.com/org/a_repo_name" => "a_repo_name"
+        "git@github.com:org/repo_2.git" => "repo_2"
+        '''
+
+        last_after_slash = self.repo_url.split('/')[-1]
```

Comment:
Do we need to include the user part as well else this can easily conflict.  The user/repo_name is what makes the url unique.  E.g. /github/james/foo /github/aldo/foo

Reply:
As discussed offline, repos are on the same organisation and also, `slugify` takes care of collisions.